src/client/boundaries.rs

```diff
@@ -40,28 +52,46 @@ impl BoundarySymbol {
 
 impl Display for BoundarySymbol {
     fn fmt(&self, f: &mut Formatter) -> Result<(), Error> {
-        if self.invisible {
-            write!(f, " ")
-        } else {
-            write!(f, "{}", self.boundary_type)
+        match self.invisible {
+            true => write!(f, " "),
+            false => match self.should_be_colored {
+                true => write!(f, "{}", self.color.paint(self.boundary_type)),
+                false => write!(f, "{}", self.boundary_type),
+            },
         }
     }
 }
 
 fn combine_symbols(
     current_symbol: BoundarySymbol,
     next_symbol: BoundarySymbol,
+    input_mode: InputMode,
 ) -> Option<BoundarySymbol> {
     let invisible = current_symbol.invisible || next_symbol.invisible;
+    let should_be_colored = current_symbol.should_be_colored || next_symbol.should_be_colored;
     let current_symbol = current_symbol.boundary_type;
     let next_symbol = next_symbol.boundary_type;
+    let color = match should_be_colored {
+        true => match input_mode {
+            InputMode::Normal => colors::GREEN,
+            InputMode::Locked => colors::GREEN,
+            InputMode::Pane => colors::WHITE,
```

Comment:
How about:
```
match input_mode {
    InputMode::Normal | InputMode::Locked => colors::GREEN,
    _ => colors::WHITE
}
```
?

Reply:
I did it this way for now, because I wasn't sure whether I would have to change colors individually for more distinction after your review